ui/shared/entities/token/TokenEntity.tsx

```diff
@@ -167,3 +168,16 @@ export {
   Content,
   Copy,
 };
+
+type NativeIconProps = Omit<IconProps, 'token'> & { src?: string | null };
+
+const NATIVE_TOKEN_DATA = {
+  name: config.chain.currency.name || '',
+  symbol: '',
+  address: '',
+  type: 'ERC-20' as const,
+};
+
+export const NativeIcon = (props: NativeIconProps) => {
```

Comment:
The component is better placed in its own file and named `NativeTokenIcon` if it is primarily used independently from `TokenEntity`. Currently, it is unclear from the contexts where `NativeIcon` is used what it actually represents.

Reply:
I think it won't be clear why we use TokenEntity for the icon and why we need that "native token data", if I move it to a separate file.

Reply:
Maybe we don't have to use `TokenIcon` then. Could be just a regular `Icon` with `TokenPlaceholder` fallback.